javascript/javascript2/week1/spirit_animal/script.js

```diff
@@ -0,0 +1,73 @@
+//Spirit animal name generator
+document.querySelectorAll('.radio').forEach(radio => {
+   radio.addEventListener('click', function(event){
+      const usernameInput = document.getElementById('username');
+      const buttonSpiritAnimal = document.getElementById('btnSpiritAnimal');
+
+      buttonSpiritAnimal.removeEventListener('click', handleButtonClick);
+      usernameInput.removeEventListener('mouseover', handleMouseOver);
+      usernameInput.removeEventListener('input', handleTextInput);
+      
+      if (event.target.value === 'btnClick') {
+         buttonSpiritAnimal.addEventListener('click', handleButtonClick);
+      } else if (event.target.value === 'hoverInput') {
+         usernameInput.addEventListener('mouseover', handleMouseOver);
+      } else if(event.target.value === 'textInput') {
+         usernameInput.addEventListener('input', handleTextInput);
+      }
+   })
+})
+
+document.getElementById('btnNewSpiritAnimal').addEventListener('click', handleNewSpiritAnimalClick);
+
+function handleButtonClick(event) {
+   handleEvent(event, 'result');
+}
+
+function handleNewSpiritAnimalClick(event) {
+   handleEvent(event, 'result2');
+}
+
+function handleMouseOver(event) {
+   handleEvent(event, 'result');
+}
+
+function handleTextInput(event) {
+   handleEvent(event, 'result');
+}
+
+const arrSpiritAnimal = [
+   'Wolf',
+   'Bear',
+   'Eagle',
+   'Butterfly',
+   'Dolphin',
+   'Owl',
+   'Tiger',
+   'Horse',
+   'Spider',
+   'Fox'
+]
+
+function getRandomSpiritAnimal() {
+   const randomIndex = Math.floor(Math.random() * arrSpiritAnimal.length);
+   return arrSpiritAnimal[randomIndex];
+}
+
+function displaySpiritAnimal(userName, spiritAnimal, elementId) {
+   const resultElement = document.getElementById(elementId);
+   resultElement.innerText = `Name: ${userName}; The Spirit Animal: ${spiritAnimal}`;
+}
+
+function handleEvent(event, elementId) {
+   event.preventDefault();
+   const userName = document.getElementById('username').value.trim();
+   if (userName === '') {
+      alert('Please, enter your name');
+   } else {
+      const randomSpiritAnimal = getRandomSpiritAnimal();
+      displaySpiritAnimal(userName, randomSpiritAnimal, elementId);
+   }
+}
```

Comment:
function names such as `DoStuff()`, `Create()`, `HandleEvent()` aren't very descriptive of what the do. Be careful about using such general names when working on larger projects. Consider instead a name such as `GenerateSpiritAnimal()`.

Besides thing one case, you're naming a very nice and descriptive. 👍